### PR TITLE
* #402: Updated the documentation about implementing a new dialect.

### LINKED ISSUE #402: Update documentation for implementing a new SQL dialect adapter
Currently the documentation for implementing a new SQL dialect adapter is not up to date, and a little bit confusing, as in https://github.com/exasol/virtual-schemas/blob/master/doc/development/developing-sql-dialect/developing_a_dialect.md

We should update it, and maybe we can use https://www.exasol.com/en/resource/an-introduction-to-exasol-virtual-schemas-the-first-part-of-our-blog-series/ and https://www.exasol.com/en/resource/how-to-build-your-own-virtual-schema-adapter-part-1/ as reference.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #402

### CODE REVIEW

File: doc/development/developing-sql-dialect/developing_a_dialect.md [hunk 1]
```diff
@@ -25,25 +25,25 @@ As an example, PostgreSQL handles some of the data types subtly different from E
...
+    | In this repository |    SQL Dialect Adapter     |   Even out specifics of the source database
+    |--------------------|----------------------------|
+    |                    |///////// JDBC API /////////|
+    |                    |----------------------------|
+    |                    |  PostgresSQL JDBC Driver   |   JDBC compliant access to payload and metadata
+    |  External          |----------------------------|
+    |                    |// PostgresSQL Native API //|
+    |                    |----------------------------|
+    |                    |         PostgreSQL         |   External data source
+    '-------------------------------------------------'
```
Comment (chiaradiamarcelo) [inline doc/development/developing-sql-dialect/developing_a_dialect.md:46]: I would remove the current PlantUML diagrams, or change them to something more "workflow" related. Take a look at the SpringMVC workflow for example, in https://tutorialspedia.com/wp-content/uploads/2019/09/Spring-MVC-Flow-Diagram.png and https://java2blog.com/wp-content/uploads/2017/07/SpringMVCTuto-1.png

We could also add a sequence diagram at some point but I think with this should be enough for now.

File: doc/development/developing-sql-dialect/developing_a_dialect.md [hunk 2]
```diff
@@ -89,7 +89,7 @@ The Java package structure of the `virtualschema-jdbc-adapter` reflects the sepa
       |     |
       |     '-- ...
       |
-      '-- jdbc                                 Base implementation for getting metadata from JDBC
+      '-- jdbc                                 Base implementation for getting metadata from JDBC (based in virtual-schema-common-jdbc repository)
```
Comment (chiaradiamarcelo) [inline doc/development/developing-sql-dialect/developing_a_dialect.md:92]: would be nice to add the links for each repository when they are mentioned, as for example:

`Base implementation for getting metadata from JDBC (based in [virtual-schema-common-jdbc repository](https://github.com/exasol/virtual-schema-common-jdbc)) `

File: doc/development/developing-sql-dialect/implementing_additional_dialect_specific_behavior.md [hunk 3]
```diff
@@ -198,8 +198,11 @@ Let's look at a HIVE dialect example. We only want to change mapping for one dat
     
         @Override
         public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
-            if (jdbcTypeDescription.getJdbcType() == Types.DECIMAL) {
+            final int jdbcType = jdbcTypeDescription.getJdbcType();
+            if (jdbcType == Types.DECIMAL) {
```
Comment (chiaradiamarcelo) [inline doc/development/developing-sql-dialect/implementing_additional_dialect_specific_behavior.md:202]: I would prefer to use a case for more readability in code, but it's just a personal taste.

File: doc/development/developing-sql-dialect/integration_testing_with_containers.md [hunk 4]
```diff
@@ -66,9 +66,9 @@ Another way to run integration tests:
 
 List of enabled integration tests:
 
-* ExasolSqlDialectIT
+* ExasolSqlDialectIT (in exasol-virtual-schema repository)
```
Comment (chiaradiamarcelo) [inline doc/development/developing-sql-dialect/integration_testing_with_containers.md:69]:
```suggestion
* ExasolSqlDialectIT (in [exasol-virtual-schema](https://github.com/exasol/exasol-virtual-schema) repository)
```

File: doc/development/developing-sql-dialect/integration_testing_with_containers.md [hunk 5]
```diff
@@ -66,9 +66,9 @@ Another way to run integration tests:
...
 * PostgreSQLSqlDialectIT
```
Comment (chiaradiamarcelo) [inline doc/development/developing-sql-dialect/integration_testing_with_containers.md:70]:
```suggestion
* PostgreSQLSqlDialectIT (in [postgresql-virtual-schema](https://github.com/exasol/postgresql-virtual-schema) repository)
```

**Review by chiaradiamarcelo — COMMENTED** [review]
Left some comments. It looks you'll have to rebase with master as the changes_4.0.5.md exists already

File: doc/development/developing-sql-dialect/developing_a_dialect.md (hunk 1 above)
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/developing_a_dialect.md:46]: We have a separate ticket for diagrams

File: doc/development/developing-sql-dialect/developing_a_dialect.md (hunk 2 above)
Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/developing_a_dialect.md:92]: Links don't work inside of this schema, I've tried it

Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/developing_a_dialect.md:97]: I've listed repositories and added links there, outside of the schemas

File: doc/development/developing-sql-dialect/developing_a_dialect.md [hunk 6]
```diff
@@ -23,30 +23,35 @@ The name SQL dialect adapter is derived from the non-standard implementation par
...
+    |                         |  PostgresSQL JDBC Driver   |   JDBC compliant access to payload and metadata
+    |  External               |----------------------------|
+    |                         |// PostgresSQL Native API //|
+    |                         |----------------------------|
+    |                         |         PostgreSQL         |   External data source
+    '------------------------------------------------------'
 
 For more information about the structure of the Virtual Schemas check the UML diagrams provided in the directory [model/diagrams](../../../model/diagrams). You either need [PlantUML](http://plantuml.com/) to render them or an editor that has PlamtUML preview built in.
 
+* [Virtual Schema Common Java Repository](https://github.com/exasol/virtual-schema-common-java)
```
Comment (chiaradiamarcelo) [inline doc/development/developing-sql-dialect/developing_a_dialect.md:52]: I suggest to use the same name as above then, either `vs-common-java-repository` or `Virtual Schema Common Java Repository`

Comment (AnastasiiaSergienko) [reply doc/development/developing-sql-dialect/developing_a_dialect.md:52]: Good Idea. Unified them

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**